add info about auth code + pkce flow (#425)
AdguardTeam/KnowledgeBaseDNS: docs/private-dns/api/overview.md

#### Request example

```bash
$ curl 'https://api.adguard-dns.io/oapi/v1/devices' -i -X GET \
    -H 'Authorization: Bearer {access_token}'
```

#### Generating access tokens with username and password

Make a POST request for the following URL with the given params to generate the `access_token`:

`https://api.adguard-dns.io/oapi/v1/oauth_token`

| Parameter      | Description                                                      |
|:---------------|:-----------------------------------------------------------------|
| **grant_type** | Must be `password`                                               |
| **username**   | Account email                                                    |
| **password**   | Account password                                                 |

> | mfa_token      | Two-Factor authentication token (if enabled in account settings) |

Reviewer: ```suggestion
| mfa_token      | Two-factor authentication token (if enabled in account settings) |
```

`https://api.adguard-dns.io/oapi/v1/oauth_token`

| Parameter           | Description                                                         |
|:--------------------|:--------------------------------------------------------------------|
| **grant_type**      | Must be `refresh_token`                                             |

> | **refresh_token**   | `REFRESH TOKEN` using which a new access token has to be generated. |

Reviewer: ```suggestion
| **refresh_token**   | `REFRESH TOKEN` used to generate a new access token. |
```

Reviewer: если правильно поняла, что имелось в виду

Location: REDIRECT_URI#access_token=...&token_type=Bearer&expires_in=3600&state=1jbmuc0m9WTr1T6dOO82
```

### Authorization Code + PKCE

:::warning

> To access this endpoint, you need to contact us at **devteam@adguard.com**.

Reviewer: ```suggestion
To access this endpoint, please contact us at **devteam@adguard.com**.
```

> Please describe the reason and use cases for this endpoint, as well as provide the redirect URI.

Reviewer: ```suggestion
In your message, please describe the reason for and use cases of this endpoint, and provide the redirect URI.
```

> Upon approval, you will receive a unique client identifier, which should be used for the **client_id** parameter.

Reviewer: ```suggestion
Once approved, you will receive a unique client identifier to use for the **client_id** parameter.
```
